DOI.js

```diff
@@ -133,12 +148,69 @@ function retrieveDOIs(DOIs, doc) {
 
 function doWeb(doc, url) {
 	var DOIs = getDOIs(doc);
-
+	Z.debug(DOIs);
+	
 	// retrieve full items asynchronously
 	retrieveDOIs(DOIs, doc);
+	
+	// or try another method (comment the line above and uncomment the two lines below)
+	/*
+	globalDOIs = DOIs;
+	retrieveDOIs2(doc);
+	*/
+	
+	// e.g. http://www.egms.de/static/de/journals/mbi/2015-15/mbi000336.shtml
 }
 
 
+function retrieveDOIs2(doc) {
+	if (globalDOIs.length==0 && remainingDOIs.length==0) {
+		completeDOIs(doc);
+	} else {
+		
+		var DOI;
+		var translate = Zotero.loadTranslator("search");
+		
+		if (globalDOIs.length==0){//DataCite
+			DOI = remainingDOIs.shift();
+			translate.setTranslator("9f1fb86b-92c8-4db7-b8ee-0b481d456428");
+		} else {//Default = CrossRef
+			DOI = globalDOIs.shift();
+			translate.setTranslator("11645bd1-0420-45c1-badb-53fb41eeb753");
+		}
+		
+		var item = {"itemType":"journalArticle", "DOI":DOI};
+		translate.setSearch(item);
+		
+		// don't save when item is done
+		translate.setHandler("itemDone", function(translate, item) {
+			if (!item.title) {
+				Zotero.debug("No title available for " + item.DOI);
+				return;
+			}
+			
+			//item.repository = "CrossRef";
+			items[DOI] = item;
+			selectArray[DOI] = item.title;
+		});
+		
+		translate.setHandler("done", function(translate) {
+			//start nested recursion again
+			retrieveDOIs2(doc);
+		});
+	
+		// Don't throw on error
+		translate.setHandler("error", function() {
+			Z.debug("Error with DOI: " + DOI);
+			remainingDOIs.push(DOI);
```

Comment:
This may lead to an infinity loop! Thus, we have to check here the case we are currently in... Okay, that is doable.